ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -1104,15 +1104,20 @@ def handle_post_reboot_test_reports(self):
 
         self.log("="*50)
 
-        self.report = {
-            "longest_downtime": (self.no_routing_stop - self.no_routing_start).total_seconds(),
-            "reboot_time": "0:00:00" if self.no_routing_stop and self.routing_always \
-                else (self.no_routing_stop - self.reboot_start).total_seconds()
-        }
-        # Add total downtime (calculated in physical warmboot test using packet disruptions)
+        if self.no_routing_stop and self.no_routing_start:
+            longest_downtime = (self.no_routing_stop - self.no_routing_start).total_seconds()
+        else:
+            longest_downtime = "N/A"
+        if self.no_routing_stop and self.reboot_start:
+            reboot_time = (self.no_routing_stop - self.reboot_start).total_seconds()
+        else:
+            reboot_time = "0:00:00"
         if 'warm-reboot' in self.reboot_type and not self.kvm_test:
-            self.report["total_downtime"] = self.total_disrupt_time
-
+            # Add total downtime (calculated in physical warmboot test using packet disruptions)
+            total_downtime = self.total_disrupt_time
+        self.report["longest_downtime"] = longest_downtime
+        self.report["reboot_time"] = reboot_time
+        self.report["total_downtime"] = total_downtime
```

Comment:
What if `reboot_type` is not `warm-reboot`? I feel in that scenario `total_downtime` would be undefined and this should be an error.

Reply:
Right, the test caught this too, and the fix is updated now.